src/DesktopTests/UIAutomation/TreeWalkers/TreeWalkerForTestTests.cs

```diff
@@ -31,13 +32,38 @@ public void BeforeEachTest()
         }
 
         [TestMethod]
-        [Timeout(1000)]
         public void RefreshTreeData_ThrowsWhenCancelled()
         {
             var cancellationToken = new CancellationTokenSource();
             cancellationToken.Cancel();
             var dataContext = new DesktopDataContext(Registrar.GetDefaultInstance(), A11yAutomation.GetDefaultInstance(), cancellationToken.Token);
             Assert.ThrowsException<OperationCanceledException>(() => _treeWalker.RefreshTreeData(TreeViewMode.Raw, dataContext));
         }
+
+        [TestMethod]
+        public void RefreshTreeData_RunsAllRules()
+        {
+            _elementMock.Setup(e => e.Framework).Returns(FrameworkId.WPF);
+            var dataContext = new DesktopDataContext(Registrar.GetDefaultInstance(), A11yAutomation.GetDefaultInstance(), CancellationToken.None);
+            _treeWalker.RefreshTreeData(TreeViewMode.Raw, dataContext);
+            var scanResults = _elementMock.Object.ScanResults;
+            Assert.AreEqual(2, scanResults.Items.Count);
+            var exclusionRule = scanResults.Items.Where(i => i.Items.FirstOrDefault().Rule.Equals(RuleId.ChromiumComponentsShouldUseWebScanner));
+            Assert.AreEqual(1, exclusionRule.Count());
+            Assert.AreEqual(Core.Results.ScanStatus.Pass, exclusionRule.FirstOrDefault().Status);
+        }
+
+        [TestMethod]
+        public void RefreshTreeData_ExcludesElementsWhenTheyViolateExclusionRules()
+        {
+            _elementMock.Setup(e => e.Framework).Returns(FrameworkId.Chrome);
+            var dataContext = new DesktopDataContext(Registrar.GetDefaultInstance(), A11yAutomation.GetDefaultInstance(), CancellationToken.None);
```

Comment:
If there a reason not to use the `DefaultContext` ?
```suggestion
            var dataContext = DesktopDataContext.DefaultContext;
```